contracts/Core/StakeManager.sol

```diff
@@ -71,23 +71,22 @@ contract StakeManager is Initializable, ACL, StakeStorage, StateManager, Pause {
         uint32 stakerId = stakerIds[msg.sender];
         if (stakerId == 0) {
             numStakers = numStakers + (1);
-            IStakedToken sToken = IStakedToken(stakedTokenFactory.createStakedToken(address(this)));
+            IStakedToken sToken = IStakedToken(stakedTokenFactory.createStakedToken(address(this), numStakers));
 
-            stakers[numStakers] = Structs.Staker(false, 0, msg.sender, address(sToken), numStakers, 10000, epoch, amount);
             // Minting
             sToken.mint(msg.sender, amount); // as 1RZR = 1 sRZR
+            stakers[numStakers] = Structs.Staker(false, 0, msg.sender, address(sToken), numStakers, 10000, epoch, amount);
             stakerId = numStakers;
             stakerIds[msg.sender] = stakerId;
         } else {
             IStakedToken sToken = IStakedToken(stakers[stakerId].tokenAddress);
             uint256 totalSupply = sToken.totalSupply();
             uint256 toMint = _convertRZRtoSRZR(amount, stakers[stakerId].stake, totalSupply); // RZRs to sRZRs
-
+            // Mint sToken as Amount * (totalSupplyOfToken/previousStake)
+            sToken.mint(msg.sender, toMint);
```

Comment:
mint is moved up to stake update, as in mint we calculate razorPutIn, which is proportional to currentStake

Reply:
external contract interaction before state update. re-entrancy vuln.

Reply:
we can't move this up
mint has to happen before stake increase as there is dependency

when we mint, in beforeTokenTransfer, we calculate razor deposited
their we need an old stake, not the new one

https://github.com/abhishekvispute/contracts/blob/52ec788e54fa8d320f0863944173485675948aa9/contracts/tokenization/StakedToken.sol#L63

Reply:
resolved